plotpal/volumes.py

```diff
@@ -0,0 +1,350 @@
+
+
+def construct_surface_dict(x_vals, y_vals, z_vals, data_vals, x_bounds=None, y_bounds=None, z_bounds=None, bool_function=np.logical_or): #not sure how to make this with self things
+    """
+    Takes grid coordinates and data on grid and prepares it for 3D surface plotting in plotly
+    
+    Arguments:
+    x_vals : NumPy array (1D) or float
+        Gridspace x values of the data
+    y_vals : NumPy array (1D) or float
+        Gridspace y values of the data
+    z_vals : NumPy array (1D) or float
+        Gridspace z values of the data
+    data_vals : NumPy array (2D)
+        Gridspace values of the data
+        
+    Keyword Arguments:
+    x_bounds : Tuple of floats of length 2
+        If specified, the min and max x values to plot
+    y_bounds : Tuple of floats of length 2
+        If specified, the min and max y values to plot
+    z_bounds : Tuple of floats of length 2
+        If specified, the min and max z values to plot
+        
+    Returns a dictionary of keyword arguments for plotly's surface plot function
+    """
+    
+        
+    if type(x_vals) == np.ndarray and type(y_vals) == np.ndarray :
+        yy, xx = np.meshgrid(y_vals, x_vals)
+        zz = z_vals * np.ones_like(xx)
+    elif type(x_vals) == np.ndarray and type(z_vals) == np.ndarray :
+        zz, xx = np.meshgrid(z_vals, x_vals)
+        yy = y_vals * np.ones_like(xx)
+    elif type(y_vals) == np.ndarray and type(z_vals) == np.ndarray :
+        zz, yy = np.meshgrid(z_vals, y_vals)
+        xx = x_vals * np.ones_like(yy)
+
+    if x_bounds is None:
+        if type(y_vals) == np.ndarray and type(z_vals) == np.ndarray and bool_function == np.logical_or :
+            x_bool = np.zeros_like(yy)
+        else:
+            x_bool = np.ones_like(yy)
+    else:
+        x_bool = (xx >= x_bounds[0])*(xx <= x_bounds[1])
+
+    if y_bounds is None:
+        if type(x_vals) == np.ndarray and type(z_vals) == np.ndarray and bool_function == np.logical_or :
+            y_bool = np.zeros_like(xx)
+        else:
+            y_bool = np.ones_like(xx)
+    else:
+        y_bool = (yy >= y_bounds[0])*(yy <= y_bounds[1])
+
+    if z_bounds is None:
+        if type(x_vals) == np.ndarray and type(y_vals) == np.ndarray and bool_function == np.logical_or :
+            z_bool = np.zeros_like(xx)
+        else:
+            z_bool = np.ones_like(xx)
+    else:
+        z_bool = (zz >= z_bounds[0])*(zz <= z_bounds[1])
+
+
+    side_bool = bool_function.reduce((x_bool, y_bool, z_bool))
+
+
+    side_info = OrderedDict()
+    side_info['x'] = np.where(side_bool, xx, np.nan)
+    side_info['y'] = np.where(side_bool, yy, np.nan)
+    side_info['z'] = np.where(side_bool, zz, np.nan)
+    side_info['surfacecolor'] = np.where(side_bool, data_vals, np.nan)
+
+    return side_info
+#turn colormesh into box with info on left side right side top, x, y and z basis, and colourmap
+class Box:
+    """
+    A struct containing information about a slice colormesh plot
+
+    # Attributes
```

Comment:
Update docstring with all relevant keyword arguments that are passed in on line 114.